proxy/http/HttpBodyFactory.cc

```diff
@@ -406,6 +414,12 @@ HttpBodyFactory::fabricate(StrList *acpt_language_list, StrList *acpt_charset_li
     set = determine_set_by_language(acpt_language_list, acpt_charset_list);
   } else if (enable_customizations == 3) {
     set = determine_set_by_host(context);
+  } else if (context->http_return_code == HTTP_STATUS_NO_CONTENT) {
```

Comment:
I just sent the request method in.

If the request was a HEAD request, then we do not want to return a response body either by default. [code](https://github.com/apache/trafficserver/blob/8dd4aa930317caa598d7622d3b7e091a4a23f875/proxy/http/HttpTransact.h#L1168)